network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/integration/NetPStateCollector.kt

```diff
@@ -46,17 +51,30 @@ class NetPStateCollector @Inject constructor(
     private val connectionQualityStore: ConnectionQualityStore,
     private val netPSettingsLocalConfig: NetPSettingsLocalConfig,
     private val netPGeoswitchingRepository: NetPGeoswitchingRepository,
-    private val netpSubscriptionManager: NetpSubscriptionManager,
     private val systemAppsExclusionRepository: SystemAppsExclusionRepository,
     private val netpVpnSettingsDataStore: NetpVpnSettingsDataStore,
+    private val subscriptions: Subscriptions,
 ) : VpnStateCollectorPlugin {
 
     override suspend fun collectVpnRelatedState(appPackageId: String?): JSONObject {
         val isNetpRunning = vpnFeaturesRegistry.isFeatureRunning(NetPVpnFeature.NETP_VPN)
         return JSONObject().apply {
             put("enabled", isNetpRunning)
             put("CGNATed", isCGNATed())
-            put("subscriptionActive", netpSubscriptionManager.getVpnStatus().isActive())
+            JSONObject().apply {
+                val active = subscriptions.getSubscriptionStatus().isActive()
+                put("isActive", active)
+                if (active) {
```

Comment:
I think it wouldn't hurt to add `hasVpnEntitlement` regardless of the subscription status. I can think of a few edge cases where those properties (sub status and entitlements) are not in sync.